### ISSUE TITLE
hazelcast-jet-sql, query hang for missing Kafka topic

### ISSUE THREAD

**Issue by neilstevenson** [hazelcast/hazelcast-jet#2695]
Using Jet 4.4-SNAPSHOT, IMDG 4.1.1-SNAPSHOT

If a definition such as 
```
String myDefinition = "CREATE EXTERNAL MAPPING something ( "
                + " neil VARCHAR "
                + ") "
                + "TYPE Kafka "
                + "OPTIONS ( "
                + " valueFormat 'json',"
                + " \"auto.offset.reset\" 'earliest',"
                + " \"bootstrap.servers\" '127.0.0.1:9092,127.0.0.1:9093,127.0.0.1:9094'"
                + ")";
```
is given, and this query
```
SELECT neil FROM something
```
is run when connectivity to Kafka exists but no topic exists with the given name (`something`)
this code will hang
```
        String query = "SELECT neil FROM something";
        SqlResult sqlResult = jetClient.getSql().execute(query);
        int count = 0;
        try {
            for (SqlRow sqlRow : sqlResult) {
                count++;
                System.out.println("sqlResult.sqlRow count " + count);
                System.out.println("sqlResult.sqlRow row " + sqlRow);
            }
        } catch (Exception e) {
            e.printStackTrace(System.out);
        }
```

**Comment by gierlachg** [comment]
When you subscribe to a non-existing Kafka topic, you will start getting messages as soon as the topic is created.
Query is not stuck (you will get similar behavior with existing topic) it's just waiting for results and because it's a streaming query it will possibly wait forever.

**Comment by neilstevenson** [comment]
Is there a way to produce a warning if the topic is missing ?

Or should the topic be auto-created ?  I know a producer has this ability (if `KAFKA_AUTO_CREATE_TOPICS_ENABLE=true` is set on the broker).

If the user types the topic name incorrectly, which is what I did, it's hard to diagnose.

**Comment by gierlachg** [comment]
> Or should the topic be auto-created

I guess this is just how Kafka works by default. You can disable automatic topic creation with `auto.create.topics.enable` set to `false` on the broker - you will get nasty NPE but that's another story.

**Comment by viliam-durina** [comment]
I tested it now, it doesn't get stuck, it just has no rows. Not sure where exactly it was stuck before, but maybe it was resolved by #2724
